### PR TITLE
Facts enhance

### LINKED ISSUE #282: Fact gathering on NCS540 - XR7 (LNT)
##### SUMMARY
ON NCS540 with an XR7 OS Variant (e.g. 7.5.2 LNT) the facts for Model and Version doesn't work, as the regex's are slightly different on these variants.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
iosxr_facts

##### ANSIBLE VERSION

```paste below
ansible [core 2.13.0]
  config file = None
  configured module search path = ['/home/weipperttadm/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/weipperttadm/pyenv/lib/python3.9/site-packages/ansible
  ansible collection location = /home/weipperttadm/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/weipperttadm/pyenv/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110]
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

```paste below
Collection  Version
----------- -------
cisco.iosxr 3.3.1
```

##### CONFIGURATION

```paste below
COLLECTIONS_PATHS(/home/weipperttadm/Develop/SCC-Transport-Backbone/Ansible/ansible.cfg) = ['/home/weipperttadm/ansible-
aap/collect>
DEFAULT_ROLES_PATH(/home/weipperttadm/Develop/SCC-Transport-Backbone/Ansible/ansible.cfg) = ['/home/weipperttadm/.ansibl
e/roles', '>
DEPRECATION_WARNINGS(/home/weipperttadm/Develop/Ansible/ansible.cfg) = False
HOST_KEY_CHECKING(/home/weipperttadm/Develop/Ansible/ansible.cfg) = False
INTERPRETER_PYTHON(/home/weipperttadm/Develop/Ansible/ansible.cfg) = /usr/bin/python3
PERSISTENT_COMMAND_TIMEOUT(/home/weipperttadm/Develop/Ansible/ansible.cfg) = 300
PERSISTENT_CONNECT_RETRY_TIMEOUT(/home/weipperttadm/Develop/Ansible/ansible.cfg) = 300
PERSISTENT_CONNECT_TIMEOUT(/home/weipperttadm/Develop/Ansible/ansible.cfg) = 300
```

##### OS / ENVIRONMENT

Debian Bullseye

##### STEPS TO REPRODUCE

Gather facts of an NCS540 system, no model and version output is given

##### EXPECTED RESULTS

        "ansible_net_model": "N540X-6Z18G-SYS-A",
        "ansible_net_version": "7.5.2 LNT",

##### ACTUAL RESULTS

no variables are filled

### PR DESCRIPTION
**Pull request by tweippert**
##### SUMMARY

Fixes #282 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

cisco.iosxr.iosxr_facts gathering

### CODE REVIEW

**Comment by KB-perByte** [comment]
Hey @tweippert Thanks for the contribution, I went through the issue you posted.
Looking at the change and the regexes that were added they seem to hinder the `network_os_model` where the difference was perhaps added for `network_os_version`. ref - [here](https://github.com/ansible-collections/cisco.iosxr/actions/runs/3219544335/jobs/5265053905)
would you please check the changes as per the use case and consider adding a Unit Test case with your contribution?
Please let me know if you need any help.
Regards

**Comment by tweippert** [comment]
@KB-perByte sure will look in this, think my regex for gathering the models are a litte too aggressive, so the XRv Model informations will also be changed. Hopefully fixed it with the new version.

For the unit test i try to figure out the necessary actions, think about the following, maybe you can give me a thumbs up or correct it:

- Adding a "show version" from a NCS 540 in the fixtures/ directory
- Adding a new test_iosxr_n540.py or similar Test File similar to the test_iosxr.py

Is this the right direction?

**Comment by KB-perByte** [comment]
> @KB-perByte sure will look in this, think my regex for gathering the models are a litte too aggressive, so the XRv Model informations will also be changed. Hopefully fixed it with the new version.
> 
> For the unit test i try to figure out the necessary actions, think about the following, maybe you can give me a thumbs up or correct it:
> 
> * Adding a "show version" from a NCS 540 in the fixtures/ directory
> * Adding a new test_iosxr_n540.py or similar Test File similar to the test_iosxr.py
> 
> Is this the right direction?

yep, that is correct we may not choose to edit the existing test files as the files are associated with one fixture.
a simple test for that new attribute should be fine.
Regards

**Review by KB-perByte — APPROVED** [review]

File: tests/unit/modules/network/iosxr/test_iosxr_n540.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#
+# (c) 2020 Red Hat Inc.
```
Comment (KB-perByte) [inline tests/unit/modules/network/iosxr/test_iosxr_n540.py:2]:
```suggestion
# (c) 2022 Red Hat Inc.
```

**Pull request merged**